crates/ruma-client-api/src/discovery/discover_homeserver.rs

```diff
@@ -104,3 +111,24 @@ impl TileServerInfo {
         Self { map_style_url }
     }
 }
+
+/// Information about a discovered authentication server.
+#[cfg(feature = "unstable-msc2965")]
+#[derive(Clone, Debug, Deserialize, Hash, Serialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct AuthenticationServerInfo {
+    /// The OIDC Provider that is trusted by the homeserver.
+    pub issuer: String,
+
+    /// The URL where the user is able to access the account management
+    /// capabilities of the OIDC Provider.
+    pub account: Option<String>,
```

Comment:
```suggestion
    #[serde(skip_serializing_if = "Option::is_none")]
    pub account: Option<String>,
```

This is required because otherwise, serde serializes `account: None` as `"account": null`, while the Matrix spec wants 'not specifying an optional field' to be omitting it, not setting it to `null`.

Reply:
Ok thanks, that makes sense. How come it isn't needed for the `pub authentication: Option<AuthenticationServerInfo>,` line too?

Reply:
Oh, that was an oversight! Bugs like that don't get discovered often because most software deals with `null` instead of the missing property just fine, plus the serialization of this response is currently only going to be visible for Conduit users.

Fixed in fa4ad103f475e6ee1d4e9170ea7f3eec930c87ed!

Reply:
Ah ha! Super, learning as I go 🙏